hwy/ops/x86_512-inl.h

```diff
@@ -5129,31 +5129,57 @@ HWY_API Vec512<int32_t> RearrangeToOddPlusEven(const Vec512<int32_t> sum0,
 
 // ------------------------------ Reductions
 
+template <class D>
+HWY_API int32_t SumOfLanesVal(D, Vec512<int32_t> v) {
+  return _mm512_reduce_add_epi32(v.raw);
+}
+template <class D>
+HWY_API int64_t SumOfLanesVal(D, Vec512<int64_t> v) {
+  return _mm512_reduce_add_epi64(v.raw);
+}
+template <class D>
+HWY_API uint32_t SumOfLanesVal(D, Vec512<uint32_t> v) {
+  return static_cast<uint32_t>(_mm512_reduce_add_epi32(v.raw));
+}
+template <class D>
+HWY_API uint64_t SumOfLanesVal(D, Vec512<uint64_t> v) {
+  return static_cast<uint64_t>(_mm512_reduce_add_epi64(v.raw));
+}
+template <class D>
+HWY_API float SumOfLanesVal(D, Vec512<float> v) {
+  return _mm512_reduce_add_ps(v.raw);
+}
+template <class D>
+HWY_API double SumOfLanesVal(D, Vec512<double> v) {
+  return  _mm512_reduce_add_pd(v.raw);
+}
+
 // Returns the sum in each lane.
 template <class D>
 HWY_API Vec512<int32_t> SumOfLanes(D d, Vec512<int32_t> v) {
-  return Set(d, _mm512_reduce_add_epi32(v.raw));
+  return Set(d, SumOfLanesVal(d, v));
```

Comment:
Nice to reuse the new SumOfLanesVal. Now that all overloads are the same, we can just have one generic template<class D> that returns/takes as an argument VFromD<D>.